packages/cli/test/integration/runners/task-runner-process.test.ts

```diff
@@ -0,0 +1,81 @@
+import { GlobalConfig } from '@n8n/config';
+import Container from 'typedi';
+
+import { TaskRunnerService } from '@/runners/runner-ws-server';
+import { TaskBroker } from '@/runners/task-broker.service';
+import { TaskRunnerProcess } from '@/runners/task-runner-process';
+import { retryUntil } from '@test-integration/retry-until';
+import { setupTaskRunnerTestServer } from '@test-integration/utils/task-runner-test-server';
+
+describe('TaskRunnerProcess', () => {
+	const authToken = 'token';
+	const globalConfig = Container.get(GlobalConfig);
+	globalConfig.taskRunners.authToken = authToken;
+	const testServer = setupTaskRunnerTestServer({});
+	globalConfig.port = testServer.port;
+
+	const runnerProcess = Container.get(TaskRunnerProcess);
+	const taskBroker = Container.get(TaskBroker);
+	const taskRunnerService = Container.get(TaskRunnerService);
+
+	afterEach(async () => {
+		await runnerProcess.stop();
+	});
+
+	const getNumConnectedRunners = () => Object.keys(taskRunnerService.runnerConnections).length;
+	const getNumRegisteredRunners = () => Object.keys(taskBroker.getKnownRunners()).length;
```

Comment:
Pretty sure this is the source of the failing tests, since this is now a Map.

Reply:
👍 looking into it